projects/plugins/boost/compatibility/page-optimize.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Compatibility for Page Optimize.
+ *
+ * @package automattic/jetpack-boost
+ */
+
+/**
+ * Handles migrating options to.
+ *
+ * @param mixed  $pre_option The value used to short-circuit retrieval using the pre_option filter.
+ * @param string $option     Name of the option.
+ *
+ * @return $pre_option       The value used to short-circuit retrieval using the pre_option filter.
+ */
+function jetpack_boost_page_optimize_migrate_options( $pre_option, $option ) {
+	// If any of the options are requested,
+	// migrate all if old ones exist.
+
+	$current_legacy_option_name = str_replace( 'jetpack_boost_', '', $option );
+	$legacy_options             = array(
+		'page_optimize-js',
+		'page_optimize-css',
+		'page_optimize-load-mode',
+		'page_optimize-js-exclude',
+		'page_optimize-css-exclude',
+	);
+	if ( ! in_array( $current_legacy_option_name, $legacy_options, true ) ) {
+		return $pre_option;
+	}
+
+	// Prevent recursion.
+	remove_filter( 'pre_option', 'jetpack_boost_page_optimize_migrate_options', 10, 2 );
+
+	// Populate new options from old options.
+	foreach ( $legacy_options as $legacy_option_name ) {
+		$old_value = get_option( $legacy_option_name );
+		if ( false !== $old_value ) {
+			update_option( 'jetpack_boost_' . $legacy_option_name, $old_value );
+			delete_option( $legacy_option_name );
+
+			// Update current option value, to use correct value (old one).
+			if ( $legacy_option_name === $current_legacy_option_name ) {
+				$pre_option = $old_value;
+			}
+		}
+	}
```

Comment:
If I'm reading this correctly, the first time anything tries to read any of the options related to pageoptimize, it will rename all pageoptimize settings to `jetpack_boost_` versions of themselves. Then, on subsequent pageloads, it will try to do the same thing again.

A few questions:
1. Do we need to rename these options? Would it make more sense to leave them unmodified?
2. If we are going to rename these options, why not do it when the module is first activated, once?

I feel like this will cause a lot of unnecessary options reads on future page-loads.

Additionally, this php file is loaded only if PageOptimize is also enabled. What happens if a user turns off PageOptimize, then turns on Boost?

Reply:
As discussed in our chat - we won't be doing any option migration for the time being.

I've removed the migration code and updated it to use the original page optimize option names in 5e203dc6970a9232135e4fa56863c73ebf5db868.